### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by buff0000n** [buff0000n/dojoarch#2]
Just a placeholder issue for people to leave feedback

**Comment by Kisutora** [comment]
Hi hi! Great work so far! :)

#### Biggest features/tweaks for me that I'd love to see that are not already in the To Do:

- Faster zoom in and out / less zoom level increments.
- Undo and Redo.
- Ability to rotate room while moving it at the same time.
- Ability to turn grid off.
- Tracker for prerequisite rooms for Halls, Labs etc.

#### Keyboard shortcuts!
- Rotate
- Duplicate
- Add new previously placed room

#### Add Room Menu
- Hover over categories to reveal room options instead of having to click.
- Should Observatory be in Interactive category?

I think making room selecting more visual would be beneficial.
Here's a quick mockup of what I'm thinking of:
![dojobuilder_concept](https://user-images.githubusercontent.com/22685232/71168648-e5bfa100-225f-11ea-9228-8e9d1cbea7d2.jpg)

**Comment by buff0000n** [comment]
@Kisutora Thanks for the feedback!
> Faster zoom in and out / less zoom level increments.
Not sure what you mean, with my mouse wheel you can already go from min to max zoom in a fraction of a second.

> Undo and Redo.
Undo/redo are always really difficult to implement, but I'll put them on the list.

> Ability to rotate room while moving it at the same time.
> Keyboard shortcuts!
> Hover over categories to reveal room options instead of having to click.
My focus right now is making something that's mobile-friendly.  I'm trying to avoid having so many keyboard shortcuts that you need an on-screen cheat sheet, or adding hover functionality that doesn't work on a touch screen.

> Ability to turn grid off.
If this is so you can take a screenshot, that's covered by this todo:
> Export as PNG

> Tracker for prerequisite rooms for Halls, Labs etc.
Covered by todos:
> Capacity/Energy tracking
> Room count constraints, for certain rooms and overall

> I think making room selecting more visual would be beneficial.
Covered by todo:
> Icons?

> Should Observatory be in Interactive category?
What's interactive in it?  The arsenal?

**Comment by Kisutora** [comment]
> Not sure what you mean, with my mouse wheel you can already go from min to max zoom in a fraction of a second.

This is what scrolling is for me.
Could be my mouse options I guess.
![giphy](https://user-images.githubusercontent.com/22685232/71207492-a751d280-22af-11ea-895b-d5cd2776e09e.gif)
![image](https://user-images.githubusercontent.com/22685232/71207615-f0098b80-22af-11ea-8a0d-f7341277b2bd.png)

> My focus right now is making something that's mobile-friendly. I'm trying to avoid having so many keyboard shortcuts that you need an on-screen cheat sheet, or adding hover functionality that doesn't work on a touch screen.

Fair enough, I respect that but I do wonder how many are going to design their dojo on mobile.
I'm not certainly going to but I'm just one person.

> What's interactive in it? The arsenal?

Yes, the arsenal and navigation. :)

**Comment by buff0000n** [comment]
> This is what scrolling is for me.

I see what's going on.  Firefox does mousewheel events different from every other browser, because of course they do.  Try it now.

> I do wonder how many are going to design their dojo on mobile.

"Mobile" also includes things like tablets and iPads.

> Yes, the arsenal and navigation. :)

I guess, but I almost never think of the Observatory as anything other than a place for decoration.  Pretty much every fully decorated Observatory I've seen has covered up the navigation spots.

**Comment by yodaxep** [comment]
This is my clan, and obstacle and elbow have a bit of an incorrect collision, since it's all normal in the game. https://buff0000n.github.io/dojocad/?v=2,298,158,0&mz=BYRgNADJ0Q+gTgU2gWgCwA4awMbKmmtAEy47QgBsY4cA9vGCgMzhU1jO4Auq74pHLxbEm-TmSaUiKcfXIgQRcSFgAbXlHRtICZCKaYOcPFJmFj6umbDFqA2MFGyQzlY+ZiA7NVn2ajjLEaFruAOZqTMwQMnJ6UTE2tLAAJgCuCUTMog5CNn62unko0jTUoqTcwGK+7KImAB41NvWwEc0uFbrAAKxgRgU58dktukhgpSVEreMjIxXxpfO2uE0AnLXlATiMBX5QrcAAlkwYoa6FqsdiFp2XuABGTBujgk+yxFhThVzp+VQHIrIT5iC5DHBNc6iTyqOgAQxqOS2XBw71KRmhbUiBQxQOaaxyK1MBUoWCgcDCAGd8dRyfF2C8oKpxugsIyJDhyAVXGSilz+JRyXzmiACdhxgKhczkPwxUz6fYQfKWUYedAuCqsAD1fFWTQ5RIUilmsxmLSAnR3sRgpwzeqgA
And keep up to good work, you're doing great!
P.S. It's my first time using Github.

**Comment by buff0000n** [comment]
@yodaxep Yep, confirmed in my dojo.  Looks like I need to shave 4m off the width of the Obstacle Course Architect room.  Thanks for the report!

**Comment by buff0000n** [comment]
@yodaxep should be fixed now.  Thanks!

**Comment by WildPenquin** [comment]
Hi!

Just my few suggestions are, mainly to facilitate collaboration:

- Annotation by: 
- arbitrary text
- automatically by room (newbies don't know what a room is just by it's outline)
- Changing colors of a room (and text)

I've found color coding can be useful to plan priorities i.e. which parts should be build first. However, if one can choose from a large selection of colors (or any RGB value), it could mean anything the user wants ;-)

Thanks anyways for this planner, seems like it has all the most important features and is the most feature-complete planner out there - and also, source available, so others can contribute if they wish!

**Comment by buff0000n** [comment]
Thanks for the feedback @WildPenquin 
> * Annotation by:
> * arbitrary text
Text is a lot harder to deal with consistently in a cross-browser way.  It also opens a whole can of formatting worms around font, size, centering, color, also annoying to deal with cross-browser.  It sounds like what you really want is Google Docs-style versioning and collaboration, and that's unfortunately way out of my league.  I'm not saying never, but not any time soon.

> * automatically by room (newbies don't know what a room is just by it's outline)
Did you know you can select a room and click on it again to see its name and stats?

> * Changing colors of a room (and text)
Unfortunately, arbitrarily tinting a grayscale image to any color is surprisingly hard in HTML5/CSS.  I think I have an idea, but it will be a while before I have the bandwidth to try it out.

**Comment by wingcobob** [comment]
hello!

similar to the guy above, this is my clan dojo as it is in game right now:
https://buff0000n.github.io/dojocad/?v=2,-147,47,0&mz=BYRgNAtAzJAsBsYAMyD6BjdkYCZls20gE5Z8lUAnAUyIhB0RQprB2JLJRCtugFZO+KKmB4I-cHm6iYsQdLA90AFxIdFLPsQ7RmaVhB1goXNhj4wo+nBizQ2StMDISpTnsEH0QCj6gATAMgcAHY-cBEAewAjEPCQpERwChVgMHlHFIxKDMF+bnMAgFc8uGS0KNyIeFC4QWyAGyjIWrB4QttGuPoocAhYMib7YwGKikbtDlgADicutV7wWBgpCzpGJwpCBzxpVCisEGPHPB4Ac0bISVOCAGciTf3VIgLT1HOHtzA36SA
As you can see, the clan hall is colliding with the great & greater halls on the bottom level. It works fine in game though ¯\\_(ツ)_/¯.

Other than that this tool has been great so far!

**Comment by fubaWoW** [comment]
hi there,
**first at all -> very great work! :)**

It would be nice if we could switch on/off to show/hide the floors above/below while in design time because "sometimes" it's very frustrating, especially if you label many rooms.

thanks in advance

**Comment by adrienne** [comment]
@buff0000n i have a suggestion wrt label colors. Currently they automatically pick up a variant of the room color and add a dark outline, and some combinations can be surprisingly hard to read. Having a separate full color picker for labels seems like overkill, but I think a couple really simple choices would improve things a lot.

First, if you could add a toggle that simply allowed a choice of white vs black outline, that would solve probably 80% of the issues right there.

Also, if you want to improve things more, have a simple selector that would allow you to choose from the following:
- the current behavior (variant of the room color) 
- 100% white text
- 100% black text
- 50% grey text

**Comment by buff0000n** [comment]
@adrienne Could you give me some examples of things that are hard to read?  Maybe in a dojocad link?  I'll see if there's something I can do to automatically pick between black/white for the outline.

**Comment by buff0000n** [comment]
@adrienne Nevermind, played around and came up with a simple rule to flip to a while outline when the brightness slider is below 25%.

FYI: you already have the ability to put a different color label on top of a room by placing a separate Label object on top of it: https://buff0000n.github.io/dojocad/index.html?v=7,-8,-31,0&mz=BYRgNADJ1SIJwC4QQCyIKyoPoBsCGMRAbAMyIBMEEyA7BmAEQAy+ARgKa6NA

**Comment by adrienne** [comment]
Oh, i didn't realize there was a label object! 

As for the outline -- you probably want to make the change happen at more like 35% brightness, not 25%, just fyi. This is above the threshold but getting hard to read (esp if i made the text smaller) 
![image](https://user-images.githubusercontent.com/89705/129044558-7796de20-efd4-4e27-86ed-79f0ba225987.png)

**Comment by adrienne** [comment]
Also @buff0000n do you take pull requests? There's a couple things i'm thinking of cleaning up/adding (like allowing display of the decoration capacity of each room)

**Comment by buff0000n** [comment]
@adrienne 

> As for the outline -- you probably want to make the change happen at more like 35% brightness, not 25%, just fyi. This is above the threshold but getting hard to read (esp if i made the text smaller)

Yeah, sure.  I'm used to working in a cave so it's easier for me to read than most people.

> Also @buff0000n do you take pull requests? There's a couple things i'm thinking of cleaning up/adding (like allowing display of the decoration capacity of each room)

Sure, but if you're thinking if putting that number in the room menu next to energy and capacity then those menus are already almost too wide for mobile.

**Comment by buff0000n** [comment]
@fubaWoW FYI: I have added something for this: Settings -> Show All Floors -> uncheck.

**Comment by Allknight96** [comment]
![image](https://user-images.githubusercontent.com/24255427/151685773-079281e4-315a-4a4b-b126-987ddf0cf941.png)
Vent Kids bash lab is larger than its tile apparently

**Comment by Nemirtingas** [comment]
Hi, I've got access to the older rooms and there is a wrong size for the old clan hall. I don't know if you really care about this as no one can be it anymore, but here's the thing:
This is our old Dojo first floor
![image](https://user-images.githubusercontent.com/9432037/163437803-a17ab6d3-63f7-4bfb-ae46-3b3d9898d374.png)

**Comment by justawarframeuser** [comment]
Amazing work! Thank you for the amazing app!

I did create my account to report a bug with the new rooms but I might as well express how your app helped my configure my Dojo with ease and test ideas before comiting to anything.

I thought you were using some data from DE to create the rooms but it looks like most of it is testing and guessing in your hands. That is really impressive.

Thank you and stay safe!

**Comment by buff0000n** [comment]
Oh, sorry about that.  Yeah, I'm unable to test the old rooms myself so they're all guesses.

Being able to fit a Bash Lab there but not a Tenno Lab is a good data point.  That narrows down that bounds to within 3 meters.

I always hesitate asking people to test room boundaries for me, because it involves either 
  (1) managing to clip your way out of the room by placing a transporter right against the ceiling (can you place decorations in these rooms?) and measuring the bounding box from the outside with waypoints
Or
  (2) building something like this
<img width="441" alt="Screen Shot 2023-12-02 at 9 23 23 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/ea0c0435-098f-4214-bcec-7218dde675c3">
Which is the only way I know how to narrow down the boundary further.

Also, I just noticed an [earlier comment](https://github.com/buff0000n/dojocad/issues/2#issuecomment-1099410107) on this ticket above that I missed about pretty much the same thing.  They have different results, but their setup is not exactly the same.  It may be that the boundary for that room is not a simple box, which makes things very hard to nail down.

**Comment by x-cessive1992** [comment]
I am building a cealing-transporter in the old clan hall now and I think I can also do the setup you showed above. 
Instead of the T-connector, I have an old Cross Connector. Does that still work?
 
 I also added you ingame, if you want to visit and maybe measure some stuff yourself :)

**Comment by buff0000n** [comment]
@x-cessive1992 Ugh, taking a look at some of the clan halls that are older but not that old, I don't think clipping out is going to help.  If the room doesn't have a skybox then there's not going to be anything outside I can visually measure.  Probably the best I can do with waypoint measuring is going to be just the interior dimensions.

> Instead of the T-connector, I have an old Cross Connector. Does that still work?
Yeah, my screenshot should have showed the old cross connector, which is what you used to get the Bash Lab result.  One thing I should add is that you don't actually have to build the Observatory in the above screenshot, just see if it's possible to build.

My schedule is pretty unpredictable, but if I see you online then I'll give you a ping.  In the meantime, before you build too much stuff, can you give me a DojoCAD link for what your dojo looks like right now?  Maybe I can work out a test plan that gives us the most bang for our time.

I really appreciate the effort, but I have to warn you that it's gonna suck.  I have a stockpile of plat in my clan vault so I usually do a lot of rushing room builds when I'm testing boundaries.

**Comment by x-cessive1992** [comment]
I was able to get out of the Clan Hall. This "T-section" at the bottom of the Clan Hall is wider than the rest.  

![grafik](https://github.com/buff0000n/dojocad/assets/152752609/b04bf8da-c828-4741-8dde-73cf4d03c217)

And this is the current state of the dojo: 
https://buff0000n.github.io/dojocad/index.html?v=2.0,-14,120,0&mz=PYCwjANAtAnBBMEAMyJiQVgFxgGzbwGYIAiAGRIH1gBjG6ODeAOg1RXQPx12PKtr1YEABxwUHTDy58K1AE7QALC3hsxqQgoCmCJczWjxaBAHZpPWVRoAPBBgNsmrVGGohEw0w8OREnCyJSORoAZwYIQh82SCRKAHNw4TAwRxM4sIioMBZvdMoQJSznPNiEgBsslNxmUuRKeV1COGy8Wpi0Bt0oDFNoavaTLRA2YWyRJEGygBcQLPglfTq4gBMAV2he6AWljq11yJadqc7gACMsqJrlymnEJUhs3F2TN2nHjBa8F8gtW0P+jBrh03P8en1skCTqCLqZHmAoXUYdANJDgfkaLD4c8Tn8LlBUd8TvAEuF0MQnj96ol+vBsVS4vFKuT+m0bkzafC2R1GWSkBTcCITrzOdBBcLKGCHmhUr5OmDTMQUmlIKC7PAlbKQZK7BgYlrXhVRcq5YzmfyZSrqUkco9COiyjSWVB7RKQN1FZtcjyCro9V63QAzNAW7xWuIgYPZOkQMOm6h0LJKIVsCQhrgyYIAKGs6uiyhTrh1ykIaSgyZcHEo5WAEFwpjLFdT8sUUDjoybmgKAEssjAhXkJNXppsUATvV3yt1-a0XigtOV6M1WXPOuULoQISIJ1WaN0vipBtl6nuqmAB6NECTT2MwIe8i6T-vWRflAhKCsVvMYJMH0UMnYt5KOi5b1LQCBgBC8A-keRRuMAACG2wItsMF-pEkrPikMBpCgJKNCGFIIg69QEdG8KIj6ZFgBgRE4nUWjUbgXzckMXT9LRYqvr87HZJxLokYx3R4IwOFykJ-TMRAMAGjxBFgFEklUm48kYFyJEqboCmjOKSLsdp0CunpqmQJ8VqaWganSbJnTyRaOSlnK+HCTGDlWteI7lo5TgsM2C4AG4lmk8CFkOiZtvmzjNhkQjoA2hhRewCjQO2oils2JLlJ5qXwA20XFr0aSJVW5Rfng-g2W45SBU8qo2YQQA

**Comment by buff0000n** [comment]
@x-cessive1992 You can stop building hallways there; you're not going to fit an observatory.  There's too much stuff in the way.

I think what you've got so far is good enough for this room.  According to your room testing, it should be strictly greater than 46m wide and less than or equal to 52m wide.  You measured the widest part at 47m, so that checks out.  You also measured the length as 65m where my screenshot measurements put it at 64.5m, so that's pretty cool.

Here's your measurements against what I've traced from screenshots:
![old-hall-test-smaller](https://github.com/buff0000n/dojocad/assets/43117210/8099e895-66fc-40f5-8a4f-aff899ed8f97) 

I think I'm just going to make it a box that's 52m wide.  I prefer to err on the side of caution.  False negatives are far less of a problem than false positives.

If you feel like doing me another favor, you can do some testing on the other hall on the floor above.  Nothing fancy, just some screenshots of what it allows you to build here:
<img width="268" alt="Screen Shot 2023-12-04 at 1 54 36 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/d4563c7c-3108-4d40-94cc-3c7f3d853dd3">

And if you feel like building yet more temporary hallways, what you can build here:
<img width="298" alt="Screen Shot 2023-12-04 at 1 55 31 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/974f4ece-0c69-4b3e-b884-914b668c0989">

And here:
<img width="306" alt="Screen Shot 2023-12-04 at 1 56 03 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/efbfe63c-4300-4766-8b3f-6598127838bc">

I just need screenshots of the Add Room screen showing what's allowed and what's not, you don't have type out a list.

Thanks!

**Comment by x-cessive1992** [comment]
@buff0000n I think we can do better than just a 52m wide box. I believe my first assumption, that the dueling room would not fit, is wrong. I think it will fit, because the wider area is only in the bottom half of the old clan hall. And I think DojoCAD should be as accurate as we can make it, to allow others to plan in the best way possible. 

Testing this only requires me to delete the temporary Reactors that I built to rearrange stuff on the main floor, so that is not a problem. 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/5eff94b6-2ecb-448e-a3ef-9249ccafb19f)

I think it might then also be interesting to test this: 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/fcc8a123-aa7d-4de8-9cca-8ddfd238d504)

And this: 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/e859ca3c-e277-4d2d-b909-5b1e51a4b17d)

What do you think - are those tests useful?

Your suggested tests for the other clan hall are also no problem.

**Comment by buff0000n** [comment]
@x-cessive1992 You're planning on deleting that hall and all your labs, too?

I would start with just seeing what will fit here
<img width="221" alt="Screen Shot 2023-12-04 at 2 28 24 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/423dd49b-d21c-47f1-9714-f87ea9b841a9">

**Comment by x-cessive1992** [comment]
Hey @buff0000n,

Yes, I plan to delete the hall and the labs. I am rebuilding almost the entire Dojo except for the legacy rooms and the 2nd floor. 

I made a mistake on the 2nd floor. The Infested Chambers are rotated differently. I am rebuilding the left one to do your requested tests there. But the current rotation is also useful, because there are some room placements that DojoCAD would allow, but cant actually be build.
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/315fbb5b-1a5d-4e63-b5e9-1c51d40427ac)

DojoCAD is wrong about Small Gardens: 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/7b29a96e-f7e0-4305-90fb-4e3eccfb0c37)

DojoCAD is wrong about the Chem Lab:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/ae2dfdb2-a070-43ab-a1a6-8f51b496f1a0)

DojoCAD is right about the halls:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/c5e480c5-f757-45a1-985e-ee92d63e195d)

DojoCAD is wrong about the Reactor:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/c28d4535-e8f6-41c3-ab28-636b859ba653)

DojoCAD is wrong about the Cross Connector and the Extended Straight Hallway:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/c6708b4b-cc05-4b3c-a09d-17cf7348b741)

DojoCAD is wrong about Courtiers Bliss and the Dueling Room
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/b4f6ce28-5b3a-44bd-9b54-db34eba55eb4)
 

I couldnt properly test the other position yet, because I forgot to delete the rooms on the bottom side.

**Comment by buff0000n** [comment]
@x-cessive1992 Okay, that narrows it down to >64m but <=70m wide.  I think that's pretty good, thanks for the details.

If you're interesting in testing if there are notches in the corners of the large halls' bounding box, like there are with the "newer" basic clan hall, then I'm having a hard time coming up with something efficient with your setup.  They best I got is this piece of shit
<img width="375" alt="Screen Shot 2023-12-04 at 10 12 16 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/0d2f1413-5255-492d-b66a-eeb895c2353e">

**Comment by x-cessive1992** [comment]
Looking at this now:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/11d4f64f-2645-414c-a235-fd1fb92ee7b8)

Nothing unusual with the gardens: 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/7198bfd1-0afe-400c-aad8-68ecc44cd1ff)

The Orokin Lab does not fit, but i guess that is no surprise now. 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/0dd406e3-c6d2-4467-9865-c7c9e278824a)

No news here:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/795e1548-0eff-4bfc-ba17-182492b78f65)

The Reactor DOES fit on the top side. It only does not fit on the bottom side (see further below) 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/c94ee1ae-8cc9-487e-8e60-f9241c8f6a24)

No news here:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/b82a6902-0a1a-4d52-9758-d9126b19319b)

As I said before, my assumption that the Dueling Room wouldnt work was wrong.
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/5fff5935-8eea-485f-9b6d-75049e155511)

Duviri Cave (and similar halls) do not work like this:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/3f465fb1-684c-40d3-b9df-90777e4f583b)

The fact that Harbingers Pass does not work might be interesting (and that Clan Tier Barrack do work):
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/fb481edc-7066-4741-89c9-6dc179eb686e)


Now properly looking at the bottom again: 
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/564c6204-772b-4fd7-9be5-e8d93e636156)

Only the Ventkids Lab and the Connectors (except Cross) work - NOTHING else:
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/149b68b6-283d-41de-b0ba-a1373062ee2f)
![grafik](https://github.com/buff0000n/dojocad/assets/152752609/89e69a4f-16b3-4ce8-88ef-cba82e88bc4f)

**Comment by buff0000n** [comment]
@x-cessive1992 
> my assumption that the Dueling Room wouldnt work was wrong.
Yeah, the dueling room is actually slightly, _very slightly_, smaller than the reactor.

So far the determining factor on the length of the upper notch is between the Barracks at 33.75m and Harbinger's Pass at 45.75m.  The .75 comes from the fact that the old cross connector extends 2.25m further than the new connectors.  

That's a pretty big range, you could try testing this to narrow it down further:
<img width="184" alt="Screen Shot 2023-12-05 at 1 16 17 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/85a17a34-17ba-4903-b25d-28eeefa34194">

**Comment by buff0000n** [comment]
We really should have started a new issue for this...

**Comment by buff0000n** [comment]
@x-cessive1992 Here's what the updated Old Hall bounds look like so far:
<img width="305" alt="Screen Shot 2023-12-05 at 3 41 15 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/989e3bd0-7390-4b99-a16a-1435624379bc">

And the Old Grand Hall, just a little bit wider:
<img width="234" alt="Screen Shot 2023-12-05 at 3 47 24 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/bd6a93e6-00bc-4317-8824-4d1bd59eca85">

If you're interested in seeing whether there's another notch at the _bottom_ of the Old Hall bounds, then you're going to have to do something like this:
<img width="501" alt="Screen Shot 2023-12-05 at 3 38 32 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/f707cc61-cc08-4e7d-8dfe-2ad95296f5a4">
If a small hallway fits there then there's a notch.  If anything else fits there then there are also notches in the Old Cross Connector's bounds.

Also, I'm curious whether Courtier's Bliss fits on the right side of the Old Hall.  It's slightly wider on one side than the other.
<img width="287" alt="Screen Shot 2023-12-05 at 3 40 55 PM" src="https://github.com/buff0000n/dojocad/assets/43117210/832fb0da-1ed8-49dd-a40d-2f000e393b4c">

Thanks so much for your help so far!

**Comment by Dilgins** [comment]
A feature I'd like to suggest when it comes to spawn points is the ability to universally set any rooms as spawn points, instead of just Clan Hall. 

For example, my clan's dry dock has a spawn point set there

**Comment by buff0000n** [comment]
@Dilgins The Arrival Gate and Spawn Room are two different things.  Even if you place an Arrival Gate, you still have to designate a Spawn Room.  It determines the structure of your Dojo and which rooms have to be deleted before deleting other rooms.  I should probably change the wording in DojoCAD to say "Spawn Room", which is what the game calls it, instead of "Spawn Point", to be less confusing, but you do still need one.

**Comment by buff0000n** [comment]
@Dilgins Okay, I think you finally pushed me over the edge to taking action.  You can now mark your Spawn Room and Arrival gate separately
<img width="571" alt="Screen Shot 2024-01-07 at 5 32 03 AM" src="https://github.com/buff0000n/dojocad/assets/43117210/f9303b46-f9bc-4488-91f5-89622162411d">

**Comment by AtSburbsEnd** [comment]
Could you add a checkbox for already built rooms so that their required resources don't affect the counter at the bottom? The "Total Resources" button only shows what's needed to build the rest but doesn't include things that have been checked off. 

Otherwise, works great and thank you so much for this tool.

**Comment by laser5mw** [comment]
Hi, I'd like to report an incorrect sizing on the Ostron Cove room. in game it connects just fine.
![230410_20250115010420_1](https://github.com/user-attachments/assets/33e4813f-af50-4b28-a406-18f9139d2089)
![230410_20250115010353_1](https://github.com/user-attachments/assets/e10d4105-21c6-4e77-a1e9-f603dc12bf2a)
On DojoCAD the Ostron Cove room is marked as invalid placement.
![wronbg](https://github.com/user-attachments/assets/d4b7fa51-db3c-4bca-8af2-dce0168e0d12)

**Comment by Chaoschaot232** [comment]
This Tool is nice but it is wrong in that point that you can build Clan Halls (all of them) only once. Correct would be no limit since we have in our dojo already the greatest clan hall and we can add another one so this needs to be fixed asap, thanks.

**Comment by buff0000n** [comment]
> This Tool is nice but it is wrong in that point that you can build Clan Halls (all of them) only once. Correct would be no limit since we have in our dojo already the greatest clan hall and we can add another one so this needs to be fixed asap, thanks.

@Chaoschaot232 I just double-checked in my dojo, and I can still only build one each of the following:
- Clan Hall
- Clan Great Hall
- Clan Greater Hall
- Clan Grand Hall
- Clan Grandest Hall

Can you show me screenshots of two of these in your Dojo at the same time?  Just shots of the map is fine.

**Comment by buff0000n** [comment]
> Hi, I was going through mapping out our dojo, and it seems like your tool doesn't have capacity added per old clan great/greater/grand/grandest halls?

Yeah, I don't know what the stats on those rooms are.  I don't have any of the discontinued rooms, so I'm dependent on other clans sending me screenshots and information.

**Comment by Litherium** [comment]
Hello! Very cool tool, thank you for making it! I mapped out my clan's dojo, which includes discontinued pieces, and I thought I'd point out that the (old) great hall allows big gardens to be placed somewhat nestled into it.

![Image](https://github.com/user-attachments/assets/6130964e-425b-4f34-92a9-e6b0cc4d7b59)

[clan layout](https://buff0000n.github.io/dojocad/?v=2.0,-25,-189,0#PYCwjANGBsELQGYBMEAMaD6BjAplWcA7NAHRICsamwWW+UAHCZeqhjXTBOS9QE54ALAE4ylME15h2A+GEJjGzKgnYBnEBEGplPKtOAb4k7lPYgUXMKnKLWEBxDUBDbABd4CHYUp700rA9oJBIfU31sTlgwcgQ7TFp4YNDfXjZE4UEU8NYMABtgJJCw4UIqJHy6OGSwpAQqNjyAEygEBTC-KHyAIwhM7IoG-IA3T29KQmFy7D54fo7eVQ56OEFohKi5MGzIdM2kHUHdyPpyEKON+gYsi-SPOHJDylX11UC5JkGXqAg3j0fFN9IH8IAxzs81j83lUwYDIcdEmAwDcIes2LI4BJAYQGD90XgHk94Di8RhZACviTjrJYZTcdShLjMdswpBpLJBLiDoo2WTGdxwT92QSuMydr8+XJYNyLqoMVwzjyJQBzPJqTyWWLxSCOCAALgAHnq9RhVXkINBBFAtYN7LqjcbsAaoGBNXFbV0sM64Ax-DazIjBFaYu60idmW74mwmi04BRov6VNg8JaoKhRF8EQTkGmM89jqr7jmQ4Ddrr9Q7TXk6KmS5nLlZE5iqLrjYaZPQJCi5BAKqqAFYa3vyUstxyGx3Klz0JAj+voVvGg1V0ap2cKecYcCdmyjtgWeiCbabkD1UVIYr53tb4MEBikMLNiogcQEGLjHv4i1WzHQB9X4VvzkBAskfXlZFTTFUEvHtAMgwgP2bQDEBQIhEOBSUULkaDsmbOVs1QsAQNwjCMRzX9-x7VQ8iCGDfWmPI8HmSgGF4ComgAV0sINGFsD12I4tkeNdY8w2AXokG0a1QwiKcOKAg5RIacslxNTi+mlVAlP8KsAFsZy0qNTTUAB3IdFKjFSJwwdTqitCyPVUEAAEsZ2gPi2IwNxNC4ZA8yGJo6HovztQwHAADNDwoeIKiwQpVlxELHPUeB41+aAZPQJAgA)

**Comment by buff0000n** [comment]
@Litherium Thanks for the correction and example layout.  I have fixed the bounds on the Old Great Hall.
